lwc/src/main/scala/quasar/physical/s3/S3LWFS.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2014–2018 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.physical.s3
+
+import quasar.Data
+import quasar.contrib.pathy._
+import quasar.mimir.LightweightFileSystem
+import slamdata.Predef._
+
+import fs2.Stream
+import org.http4s.client.Client
+import org.http4s.Uri
+import scalaz.concurrent.Task
+
+final class S3LWFS(jsonParsing: S3JsonParsing, uri: Uri, client: Client) extends LightweightFileSystem {
+
+  def children(dir: ADir): Task[Option[Set[PathSegment]]] = impl.children(client, uri, dir)
+
+  def read(file: AFile): Task[Option[Stream[Task, Data]]] = impl.read(jsonParsing, client, uri, file)
+
+  def exists(file: AFile): Task[Boolean] = impl.exists(client, uri, file)
```

Comment:
This will be more efficient, that's the only thing; much less data in the `HEAD` response than getting all children, especially in some cases because S3 listings are actually recursive. It also happens to be really simple. Good to know, though.

Reply:
:+1: 

Also I don't have a branch pushed removing the requirement for `exists` to be implemented yet anyways.